treeherder/etl/resultset_loader.py

```diff
@@ -17,34 +17,34 @@ class ResultsetLoader:
     """Transform and load a list of Resultsets"""
 
     def process(self, message_body, exchange):
+        transformer = self.get_transformer_class(exchange)(message_body)
         try:
-            transformer = self.get_transformer_class(exchange)(message_body)
-            try:
-                repo = Repository.objects.get(url=transformer.repo_url,
-                                              branch=transformer.branch,
-                                              active_status="active")
-
-            except ObjectDoesNotExist:
-                repo_info = message_body.get("details",
-                                             message_body["payload"])
-                newrelic.agent.record_custom_event("skip_unknown_repository",
-                                                   repo_info)
-                logger.warn("Skipping unsupported repo: {} {}".format(
-                    transformer.repo_url,
-                    transformer.branch))
-                return
-
-            transformed_data = transformer.transform(repo.name)
-
-            logger.info("Storing resultset for {} {} {}".format(
-                repo.name,
+            repo = Repository.objects.get(url=transformer.repo_url,
+                                          branch=transformer.branch,
+                                          active_status="active")
+
+        except ObjectDoesNotExist:
+            repo_info = message_body.get("details",
+                                         message_body.get("payload",
+                                                          message_body))
```

Comment:
I was still confused why this third case is here (`message_body`), but looking at the doc strings in this file I see that `details` is for GitHub pushes and `payload` for Hg pushes, so this fallback is for things that are neither? Given the `get_transformer_class` would raise in that case, do we still need it?

I'm wondering whether the transformer classes themselves should have `get_metadata` methods that know how to return the correct properties, which are then called here?